lib/proxy/validations.rb

```diff
@@ -11,20 +18,42 @@ def valid_mac?(mac)
     false
   end
 
+  def valid_ip?(ip)
+    !!IPAddr.new(ip) rescue false
+  end
+
+  def valid_ip4?(ip)
+    Resolv::IPv4::Regex.match?(ip)
+  end
+
+  def valid_ip6?(ip)
+    Resolv::IPv6::Regex.match?(ip)
+  end
+
   # validates the ip address
   def validate_ip(ip)
```

Comment:
A new argument is a good idea, compatible way of preparing this for IPv6. I would use `version: 4` as the default argument not to break existing DHCP modules tho.

Reply:
> Accepting IPv6 may lead to problems.

Can you elaborate? Not following?

Reply:
> > Accepting IPv6 may lead to problems.
> 
> Can you elaborate? Not following?

If today something calls `validate_ip` and expects a valid IPv4 address, but the guarantee is reduced to only a valid IP address, then code later on may fail. For example, I'm not sure how well our DHCP API would work with IPv6. As the API user it would likely change from HTTP 400 Bad Request to HTTP 500 Internal Server Error.